secretsstoresynccontroller/api/v1alpha1/secretstoresync_types.go

```diff
@@ -0,0 +1,147 @@
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// Annotations contains key-value pairs representing annotations associated with the Kubernetes secret object.
+	// The following annotation prefix is reserved: secrets-store.sync.x-k8s.io/.
+	// Creation fails if the annotation is specified in the SecretSync object by the user.
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+// SecretSyncSpec defines the desired state for synchronizing secret.
+type SecretSyncSpec struct {
+	// SecretProviderClassName specifies the name of the secret provider class used to pass information to
+	// access the secret store.
+	// +kubebuilder:validation:Required
+	SecretProviderClassName string `json:"secretProviderClassName"`
+
+	// ServiceAccountName specifies the name of the service account used to access the secret store.
+	// +kubebuilder:validation:Required
+	ServiceAccountName string `json:"serviceAccountName"`
+
+	// SecretObject specifies the configuration for the synchronized Kubernetes secret object.
+	// +kubebuilder:validation:Required
+	SecretObject SecretObject `json:"secretObject"`
+
+	// ForceSynchronization can be used to force the secret synchronization.
+	// This provides a mechanism to trigger a secret synchronization, for example if the secret hash is the same and
+	// the user requires a secret update. The string is not used for any other purpose than to trigger a secret
+	// synchronization.
+	// This field is not used to resolve synchronization conflicts.
+	// It is not related with the force query parameter in the Apply operation.
+	// https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts
+	// +optional
+	ForceSynchronization string `json:"forceSynchronization,omitempty"`
+}
+
+// SecretSyncStatus defines the observed state of the secret synchronization process.
+type SecretSyncStatus struct {
+	// SecretObjectHash contains the hash of the secret object data, data from the SecretProviderClass (e.g. UID,
+	// apiversion, name, namespace, parameters), and similar data from the SecretSync. This hash is used to
+	// determine if the secret changed.
+	// The hash is calculated using the HMAC (Hash-based Message Authentication Code) algorithm, using bcrypt
+	// hashing, with the SecretsSync's UID as the key.
+	// 1. If the hash is different, the secret is updated.
+	// 2. If the hash is the same, the secret is still updated when:
+	//		1. The LastRetrievedTimestamp is older than the current time minus the
+	//			rotationPollInterval, the secret is updated.
+	// 		2. The ForceSynchronization is set, the secret is updated.
+	//		3. The SecretUpdateStatus is 'Failed', the secret is updated.
+	// +optional
+	SecretDataObjectHash string `json:"secretDataObjectHash,omitempty"`
+
+	// LastRetrievedTimestamp represents the last time the secret was retrieved from the Provider and updated.
+	// +optional
+	LastRetrievedTimestamp *metav1.Time `json:"lastRetrievedTimestamp,omitempty"`
```

Comment:
if this means the last time we successfully retrieved *and* wrote the secret, I would call this `lastSuccessfulSyncTime`